drivers/rtc/rtc_mc146818.c

```diff
@@ -583,26 +506,30 @@ struct rtc_driver_api rtc_mc146818_driver_api = {
 #endif /* CONFIG_RTC_UPDATE */
 };
 
-static int rtc_mc146818_init(const struct device *dev)
-{
-	IRQ_CONNECT(DT_INST_IRQN(0),
-			DT_INST_IRQ(0, priority),
-			rtc_mc146818_isr, NULL,
-			DT_INST_IRQ(0, sense));
-	irq_enable(DT_INST_IRQN(0));
-
-	return 0;
-}
+#define RTC_MC146818_INIT_FN_DEFINE(n)						\
+	static int rtc_mc146818_init##n(const struct device *dev)		\
+	{									\
+		rtc_write(RTC_REG_A, 0);					\
```

Comment:
Clearing RTC_REG_A is blocking RTC time update on real hardware.

Reply:
I see, the QEMU board has no oscillator connected to the RTC (the clock therefore runs regardless of divider settings), my theory is that your RTC is connected to an oscillator. Could you check if setting the DV2, DV1 and DV0 bits according to the RTC's clock input? If that works, I will expand the dts and driver with input clock support :)

Reply:
Yes, setting DV2-0 according to RTC clock allows it run. Freq on my hardware is 32.786KHz so DV sequence is 0-1-0.

Reply:
I have added input clock frequency configuration to the driver and devicetree (See [commit](https://github.com/zephyrproject-rtos/zephyr/pull/60321/commits/f33e973278b587cd764916181a1f34ffc9a51ad7)). Please test it out :)

Reply:
It is working👍

Reply:
Awesome :)